hypothesis-python/src/hypothesis/extra/django/_fields.py

```diff
@@ -262,6 +262,51 @@ def _for_form_boolean(field):
     return st.booleans()
 
 
+def _model_choice_strategy(field):
+    def _strategy():
+        if field.choices is None:
+            # The field was instantiated with queryset=None, and not
+            # subsequently updated.
+            raise InvalidArgument(
+                "Cannot create strategy for ModelChoicesField with no choices"
+            )
+        elif hasattr(field, "_choices"):
+            # The choices property was set manually.
+            choices = field._choices
+        else:
+            # choices is not None, and was not set manually, so we
+            # must have a QuerySet.
+            choices = field.queryset
+
+        if not choices.ordered:
+            raise InvalidArgument("QuerySet must have explicit ordering")
+
+        return st.sampled_from(
+            [
+                (
+                    choice.value
+                    if isinstance(choice, df.models.ModelChoiceIteratorValue)
+                    else choice  # Empty value, if included.
+                )
+                for choice, _ in field.choices
+            ]
+        )
+
+    # Accessing field.choices causes database access, so defer the strategy.
+    return st.deferred(_strategy)
+
+
+@register_for(df.ModelChoiceField)
+def _for_model_choice(field):
+    return _model_choice_strategy(field)
+
+
+@register_for(df.ModelMultipleChoiceField)
+def _for_model_multiple_choice(field):
+    min_size = 1 if field.required else 0
+    return st.sets(_model_choice_strategy(field), min_size=min_size).map(list)
```

Comment:
```suggestion
    return st.lists(_model_choice_strategy(field), min_size=min_size, unique=True)
```
Set iteration order is nondeterministic, so that would cause flaky tests if they happened to be order-dependent - but we can avoid that by generating a unique list directly 🙂

Reply:
Nice, I missed the `unique` option.